File: virt-install_usage.pl
#!/usr/bin/perl
foreach (<DATA>)
{
	print $_;
}
__DATA__


qemu-img create -f qcow2   /vmstorage/Centos7.img  80G 

virt-install -n Centos7 -r 2048     -cpus=2  -f /vmstorage/Centos7.img


virt-install --name=centos7a --ram=1024 --vcpus=1 --cdrom /vmstorage/CentOS-7-x86_64-DVD-1503-01.iso  \
    --disk path=/vmstorage/centos7a.img,size=70 --network bridge=br-ceph-linux --vnc --vncport=5900 \
    --vnclisten=0.0.0.0 --os-type=linux --os-variant=rhel7 --connect qemu:///system  


#这种安装方式注意  sda  -> vda了 
# /root/bin/kickstart/centos7_1_x64.cfg  的. ignoredisk --only-use=sda 这行也就错误了. 
# 
virt-install --vnc --noautoconsole --name=server --ram=2048 --arch=x86_64 --vcpus=1 --os-type=linux --os-variant=rhel7 \
--hvm --accelerate --disk=/vmstorage/server.img,size=80 --location=http://mirrors.aliyun.com/centos/7.2.1511/os/x86_64/  \
--extra-args="ks=http://192.168.1.11/ks.cfg" && virt-install --autostart server


virt-install --vnc \
                --noautoconsole \
                --name=server$i \
                --ram=512 --arch=x86_64 \
                --vcpus=1 --os-type=linux \
                --os-variant=rhel7 \
                --hvm --accelerate \
                --disk=/var/lib/libvirt/images/server$i.img size=8G \
                --location=ftp://192.168.0.254/pub/rhel6/dvd \
                --extra-args="ks=http://192.168.0.158/ks.cfg"  

File: virt_install_usage.py
#!/usr/bin/python
#coding=utf-8

import os
import sys;

if len(sys.argv) != 2:
    print "%s  input.iso"%(sys.argv[0]);
    sys.exit(0);

iso=sys.argv[1];
name=os.path.basename(iso)
#name="centos7"



#  bin/develop_python/libvirt-list_network.py
#--network network=ovs-net,portgroup=vlan59 \\
DATA="""
#--network network:br-ex
#也可以采用virt-install命令下直接产生虚机
/usr/bin/virt-install \\
        --name {name} \\
        --ram 1024 \\
        --virt-type kvm \\
        --vcpus 1 \\
        --disk path=/vmstorage/{name}.img,size=20 \\
        --cdrom {iso} \\
        --graphics vnc,password=123456,listen=0.0.0.0 \\
        --network network=mybridge166 \\
        --force \\
        --autostart 
""".format(iso=iso, name=name);
print DATA;

